### PR TITLE
[fix] BaseChildFormSet doesn't caught IntegrityError

### PR DESCRIPTION
**Pull request by WnP**
This PR also fix wagtail/wagtail#2136

### CODE REVIEW

File: modelcluster/forms.py [hunk 1]
```diff
@@ -127,6 +128,19 @@ def save(self, commit=True):
 
         return saved_instances
 
+    def clean(self, *args, **kwargs):
+        if any(self.errors):
+            return
```
Comment (WnP) [inline modelcluster/forms.py]: Don't bother validating the formset unless each form is valid on its own

File: modelcluster/forms.py [hunk 2]
```diff
@@ -127,6 +128,19 @@ def save(self, commit=True):
...
+        for entry in self.queryset:
+            try:
+                entry.full_clean()
+            except ValidationError as e:
+                e.error_list = e.error_dict.get('__all__', [
+                    ValidationError('Integrity on {}'.format(self.model))
```
Comment (WnP) [inline modelcluster/forms.py]: Provide a default Validation error

**Comment by gasman** [comment]
Thanks @WnP! Could you add a test for this, please?

**Comment by WnP** [comment]
@gasman sure, I've added a test and also rebased this branch

File: tests/tests/test_cluster_form.py [hunk 3]
```diff
@@ -5,7 +5,7 @@
 from django.test import TestCase
 from tests.models import Band, BandMember, Album, Restaurant, Article, Author, Document, Gallery
 from modelcluster.forms import ClusterForm
-from django.forms import Textarea, CharField, ModelForm
+from django.forms import Textarea, CharField
```
Comment (WnP) [inline tests/tests/test_cluster_form.py]: `ModelForm` was imported but not used -maybe you prefer another PR for that, let me know

**Comment by WnP** [comment]
Hi @gasman,

Thanks for pointing me out [this part of django's code](https://github.com/django/django/blob/b99221b8bb2a4cd7fd5ac6e5f03e8eb1953d6384/django/forms/models.py#L691-L714) it's really cleaner than my solution.

I've implemented this feature using it, which is almost a copy of the original code.

Also thanks for the other stylistic advices

**Comment by gasman** [comment]
Hi @WnP - thanks for the update. I'm not seeing any changes on this branch - have you pushed it?

**Comment by WnP** [comment]
Hi @gasman,

Sorry, my bad I forgot to push, here it is

**Comment by gasman** [comment]
Merged in 39349d973bf25e6c971958df279f0d16bedf9923 - thanks @WnP!

(Tests were failing against Django 1.9 due to unique_fields not being defined there, so I've taken the easy and sensible route and dropped Django 1.9 support :-) )

**Pull request merged**